vr-wordle/src/App.test.js

```diff
@@ -1,20 +1,474 @@
+    setReset(!state.reset);
+  };
+
+  expect(state.reset).toBeFalsy();
+
+  resetPositions();
+
+  expect(state.reset).toBeTruthy();
+
+  resetPositions();
+
+  expect(state.reset).toBeFalsy();
+});
+
+test("deleteOldGuess correctly deletes the children of the target object", () => {
+  let currentGuess = ["w", "r", "d", "l", "e"];
+
+  const answer = "tests";
+
+  let letters = { current: { children: [] } };
+
+  letters.current.children = generateLetters(false, alphabet, "letters");
+
+  expect(letters.current.children).toHaveLength(26);
+
+  const deleteOldGuess = () => {
+    for (const letter of currentGuess) {
+      if (!answer.split("").includes(letter)) {
+        const indexToRemove = letters.current.children.findIndex((child) => {
+          return child.children[0].name === letter;
+        });
+        if (indexToRemove !== -1) {
+          letters.current.children.splice(indexToRemove, 1);
+        }
+      }
+    }
+  };
+
+  deleteOldGuess();
+  expect(letters.current.children).toHaveLength(22);
+  currentGuess = ["c", "h", "o", "k", "e"];
+  deleteOldGuess();
+  expect(letters.current.children).toHaveLength(18);
+  currentGuess = ["t", "e", "s", "t", "s"];
+  deleteOldGuess();
+  expect(letters.current.children).toHaveLength(18);
+});
+
+test("setGuess correctly updates the new guess", () => {
+  let state = { currentGuess: [] };
+
+  const setCurrentGuess = (array) => {
+    state.currentGuess = array;
+  };
+
+  const setGuess = (char, i) => {
+    if (char && state.currentGuess[i] !== char) {
+      const dummyArr = state.currentGuess;
+      dummyArr[i] = char;
+      setCurrentGuess(dummyArr);
+    }
+  };
+
+  setGuess("a", 3);
+  expect(state.currentGuess[3]).toBe("a");
+  setGuess("b", 3);
+  expect(state.currentGuess[3]).toBe("b");
+  setGuess("a", 0);
+  setGuess("a", 1);
+  setGuess("a", 2);
+  setGuess("a", 4);
+
+  expect(state.currentGuess.join("")).toMatch("aaaba");
+});
+
+test("submitGuess correctly submits guess, and deletes incorrect letters", () => {
+  let state = {
+    guessCount: 0,
+    currentGuess: ["a", "a", "a", "a", "a"],
+    gameEnd: false,
+    guesses: ["     ", "     ", "     ", "     ", "     ", "     "],
+    answer: "tests",
+  };
+
+  const setGuesses = (array) => {
+    state.guesses = array;
+  };
+
+  const setGuessCount = (count) => {
+    state.guessCount = count;
+  };
+
+  const setGameCondition = (game) => {
+    state.gameEnd = game;
+  };
+
+  let letters = { current: { children: [] } };
+
+  letters.current.children = generateLetters(false, alphabet, "letters");
+
+  const deleteOldGuess = () => {
+    for (const letter of state.currentGuess) {
+      if (!state.answer.split("").includes(letter)) {
+        const indexToRemove = letters.current.children.findIndex((child) => {
+          return child.children[0].name === letter;
+        });
+        if (indexToRemove !== -1) {
+          letters.current.children.splice(indexToRemove, 1);
+        }
+      }
+    }
+  };
```

Comment:
You repeat this function a couple of times, maybe we could move this out of the functions into the bigger scope?